node/coinstacks/solana/api/src/app.ts

```diff
@@ -45,4 +58,32 @@ app.get('/', async (_, res) => {
 
 app.use(middleware.errorHandler, middleware.notFoundHandler)
 
-app.listen(PORT, () => logger.info('Server started'))
+const server = app.listen(PORT, () => logger.info('Server started'))
+
+const transactionHandler: TransactionHandler<GeyserResultTransaction, GeyserResultTransaction> = async (geyserTx) => {
+  const addresses = geyserTx.transaction.message.accountKeys.map((key) => key.pubkey)
+
+  return { addresses, tx: geyserTx }
+}
+
+const registry = new Registry({
+  addressFormatter: (address: string) => address,
+  blockHandler: async () => {
+    // @TODO: What should we do with this?
+    return { txs: [] }
```

Comment:
Not sure we need it with solana tbh, I need to investigate why we does this with other chains, I guess it's to detect new transactions inside new blocks

Reply:
I took some time investigating this:

- We use the block websocket to retrieve all the transactions inside new blocks
- `blockSubscribe` is not implemented on helius endpoints
- `slotSubscribe` can be used as a fallback but it does includes the slot number only, getting all the transactions for a block would require an extra RPC call which would cost a lot in the end regarding any block validation is taking under 0.4s 

Can we skip that for Solana as the transaction geyser seems stable enough?

Reply:
The previous notion for websockets was the `subscribeTxs` (pending mempool) and `subscribeBlocks` (confirmed). We aren't actually using or forwarding block events to the client so considering the geyser websocket setup and being able to subscribe to confirmed/finalized and the notion that pending doesn't serve much of a purpose in the context of solana, you are correct that the block handler is not necessary.